circuit/src/execution-delivery/src/exec_composer/mod.rs

```diff
@@ -849,4 +892,335 @@ mod tests {
             )
         });
     }
+
+    #[test]
+    fn pre_run_bunch_until_break_multiple_contracts_succeeds() {
```

Comment:
```suggestion
    fn pre_run_bunch_until_break_succeds_for_both_contracts() {
```

Reply:
pre_run_bunch_until_break_succeeds_for_two_contracts

I have used this. It feels important to say that only 2 contracts are being tested.
Resolving this.